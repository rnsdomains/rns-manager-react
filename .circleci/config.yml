version: 2
jobs:
  build:
    working_directory: ~/tmp
    docker:
      - image: circleci/node:12
      - image: trufflesuite/ganache-cli
        command: ganache-cli --deterministic
    steps:
      - checkout
      - run: yarn
      - run: yarn test
