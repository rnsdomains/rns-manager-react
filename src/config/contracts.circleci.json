{"rns":"0xf6663c05d693b80057d6a7d9ce68f777e356650d","reverseRegistrar":"0x6a8db48ca2cbf03dd5d9d4d6517a440630acf487","publicResolver":"0x4e0585719f720a6363ebfcb1a0f5dd2b0eb1b661","nameResolver":"0x8507ceebae3bb62ac44d3362448a2e2517068126","multiChainResolver":"0x04ff4ed951229268b240eb37af782dad006510d4","rif":"0x625da57679846178b1211dbd72b25f3e07987e29","fifsRegistrar":"0xf6d4a57eeda04f1313c09cee98f6a2271e873343","fifsAddrRegistrar":"0xad4cb11e2fc5b953b47fd5d630cd40cf4ee359c1","rskOwner":"0x3a850a3f923721f2452463fe493c42fc94a2b7a3","renewer":"0x579d31ebb08ac27206dc865feccfce72d0b391fd"}