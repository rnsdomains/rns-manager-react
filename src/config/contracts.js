const RSK_MAINNET_CHAIN_ID = 30;
const RSK_TESTNET_CHAIN_ID = 31;

const RSK_MAINNET = {
  name: "RSK MainNet",
  rns: "0xcb868aeabd31e2b66f74e9a55cf064abb31a4ad5",
  registrar: "0x5269f5bc51cdd8aa62755c97229b7eeddd8e69a6",
  publicResolver: "0x4efd25e3d348f8f25a14fb7655fba6f72edfe93a",
  multiChainResolver: "0x99a12be4c89cbf6cfd11d1f2c029904a7b644368",
  rif: "0x2acc95758f8b5f583470ba265eb685a8f45fc9d5",
  revealPeriod: 259200
};

const RSK_TESTNET = {
  name: "RSK TestNet",
  rns: "0x83355fcb41acbe3919e4ff73ecffc07a3147b7e8",
  registrar: "0xb0cf0517302acf52f967d0342827ff9c01d353f2",
  publicResolver: "0xf1143e2797bef4f8bd6059605e8134686efaa355",
  multiChainResolver: "0x0",
  rif: "0xd8c5adcac8d465c5a2d0772b86788e014ddec516",
  revealPeriod: 259200
};

export default (location) => {
  if (window && window.web3 && window.web3.version) {
    window.web3.version.getNetwork((err, netId) => {
      switch (Number(netId)) {
        case RSK_TESTNET_CHAIN_ID:
          console.log("testnet", location);
          return RSK_TESTNET;

        case RSK_MAINNET_CHAIN_ID:
          console.log("mainnet", location);
          return RSK_MAINNET;
        default:
          console.log("unknown", location);
          return { name: "UNKNOWN" };
      }
    });
  }
  console.log("notweb3", location);
  return { name: "NO WEB3" };
};
