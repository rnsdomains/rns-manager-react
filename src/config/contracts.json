{
  "rns": "0x3AdE9908F7Ccb088aDe5a0e0cC5e5Df6EeD7529f",
  "registrar": "0x5f85f3CcD01b43B250b881C55644e1D0B71D4338",
  "fifsRegistrar": "0xb137FB20a8b0b9dA0A05F8F8418959fF517032F2",
  "rskOwner": "0xa8D50e7f9e930F20ceBeA7C942346fB25Db0C505",
  "rif": "0x215Caa8b09FfE3617D7B4f54FD1A32985bC388ce",

  "reverseRegistrar": "0x01C555768dBa88605615a40c439D1Cdc73Bc81d0",
  "publicResolver": "0x84204C66ad5C217f0CA49F9b694deF0617Eaac02",
  "nameResolver": "0x55f87a9EB8bB5afD2B30b17f8BE61a3f9f86eBc4",
  "multiChainResolver": "0x21871C2689d48f275F8B3ea6F2AC6c8c7A381Bfc",
  "revealPeriod": 172800
}
