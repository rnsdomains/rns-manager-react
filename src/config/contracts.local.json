{
  "rns": "0x6536263cc0aa7f24d890Ce979D3F3D488bF0eC2D",
  "registrar": "0x5ee1FE294eBF4e0fEE6eBa08c915056Fb9692a79",
  "reverseRegistrar": "0x8970ce36034133420c0Ff574f67a49679E36c6DF",
  "publicResolver": "0x76b0634FC077382240812b4F6f21D547251348FB",
  "nameResolver": "0x347220b3C5994E4B712d98a0FcE4b73085c8035e",
  "multiChainResolver": "0x51536B1Bb38703F7569C54e97eAAaf150b3D9861",
  "rif": "0xd61a3CCce5501F01D57c487A1CFE433a1c684568",
  "fifsRegistrar": "0x0613c919E4af2068C671B8788a1288f14B5c9F0C",
  "rskOwner": "0x3528149D45A110bC21b7B71AC4dA7F6C0049C854",
  "renewer": "0x93e1d7714f604d967fB56D312b980695d9A09FaD"
 }