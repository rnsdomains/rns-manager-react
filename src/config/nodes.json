{
  "rskMain": "http://localhost:7545"
}
