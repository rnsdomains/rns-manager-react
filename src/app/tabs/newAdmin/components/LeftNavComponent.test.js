import React from 'react';
import { HashRouter } from 'react-router-dom';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { mockStoreEnglish } from '../../../../../tests/config/mockStore';

import LeftNavComponent from './LetftNavComponent';

const store = mockStoreEnglish();

describe('LeftNavComponent', () => {
  it('matches snapshot', () => {
    const component = mount(
      <Provider store={store}>
        <HashRouter>
          <LeftNavComponent location="/newAdmin" advancedView={false} domain="foobar.rsk" />
        </HashRouter>
      </Provider>,
    );
    expect(component).toBeDefined();
  });

  it('shows all items when advancedView is true', () => {
    const component = mount(
      <Provider store={store}>
        <HashRouter>
          <LeftNavComponent location="/newAdmin/subdomains" advancedView domain="foobar.rsk" />
        </HashRouter>
      </Provider>,
    );
    expect(component.find('li').length).toBe(6);
  });

  it('sets correct item active when passed', () => {
    const component = mount(
      <Provider store={store}>
        <HashRouter>
          <LeftNavComponent location="/newAdmin/subdomains" advancedView={false} domain="foobar.rsk" />
        </HashRouter>
      </Provider>,
    );

    expect(component.find('a.active').text()).toEqual('subdomains');
    expect(component.find('li').length).toBe(3);
  });

  it('sets home as active when resolver is passed, but advancedView is false.', () => {
    const component = mount(
      <Provider store={store}>
        <HashRouter>
          <LeftNavComponent location="/newAdmin/resolver" advancedView={false} domain="foobar.rsk" />
        </HashRouter>
      </Provider>,
    );

    expect(component.find('a.active').text()).toEqual('Domain info');
  });
});
