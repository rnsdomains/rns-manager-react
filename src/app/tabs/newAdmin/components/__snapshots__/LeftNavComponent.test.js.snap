// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeftNavComponent matches snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <HashRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(LeftNavComponent)
        advancedView={false}
        location="/newAdmin"
        logOut={[MockFunction]}
      >
        <LeftNavComponent
          advancedView={false}
          currentLanguageCode="en"
          dispatch={[Function]}
          location="/newAdmin"
          logOut={[MockFunction]}
          strings={
            Object {
              "admin": "admin",
              "advanced": "Advanced",
              "basic": "Basic",
              "domain_info": "Domain info",
              "log_out": "log out",
              "resolver": "resolver",
              "subdomains": "subdomains",
              "your_addresses": "Your addresses",
            }
          }
        >
          <ul>
            <li>
              <Link
                className="active"
                to="/newAdmin"
              >
                <LinkAnchor
                  className="active"
                  href="#/newAdmin"
                  navigate={[Function]}
                >
                  <a
                    className="active"
                    href="#/newAdmin"
                    onClick={[Function]}
                  >
                    Domain info
                  </a>
                </LinkAnchor>
              </Link>
            </li>
            <li>
              <Link
                className=""
                to="/newAdmin/addresses"
              >
                <LinkAnchor
                  className=""
                  href="#/newAdmin/addresses"
                  navigate={[Function]}
                >
                  <a
                    className=""
                    href="#/newAdmin/addresses"
                    onClick={[Function]}
                  >
                    Your addresses
                  </a>
                </LinkAnchor>
              </Link>
            </li>
            <li>
              <Link
                className=""
                to="/newAdmin/subdomains"
              >
                <LinkAnchor
                  className=""
                  href="#/newAdmin/subdomains"
                  navigate={[Function]}
                >
                  <a
                    className=""
                    href="#/newAdmin/subdomains"
                    onClick={[Function]}
                  >
                    subdomains
                  </a>
                </LinkAnchor>
              </Link>
            </li>
            <li>
              <button
                onClick={[MockFunction]}
                type="button"
              >
                log out
              </button>
            </li>
          </ul>
        </LeftNavComponent>
      </Connect(LeftNavComponent)>
    </Router>
  </HashRouter>
</Provider>
`;
